File: n03-practice11/src/ru/mirea/n03pr11/model/ReaderSub.java
package ru.mirea.n03pr11.model;

public class ReaderSub {
    private Long num;
    private String name;
    private String surname;
    private String secondName;
    private String address;

    public ReaderSub(Long num, String name, String surname, String secondName, String address) {
        if (countLength(num) != 5)
            throw new IllegalArgumentException("Длина номера должна равняться 5");
        this.num = num;
        this.name = name;
        this.surname = surname;
        this.secondName = secondName;
        this.address = address;
    }

    private int countLength(Long num) {
        return String.valueOf(Math.abs(num)).length();
    }

    public Long getNum() {
        return num;
    }

    @Override
    public String toString() {
        return "{" +
                "name = '" + name + '\'' +
                ", surname = '" + surname + '\'' +
                ", secondName = '" + secondName + '\'' +
                ", address = '" + address + '\'' +
                '}';
    }
}

File: n04-practice12/reader_sub.cpp
//
// Created by akimovve on 19.10.2020.
//

#include <iostream>
#include "reader_sub.h"

using namespace std;

/**
 * Реализация класса "Читательский абонемент"
 */
ReaderSub::ReaderSub(string num, string name, string surname, string second_name, string address) {
	this->num = num;
	this->name = name;
	this->surname = surname;
	this->second_name = second_name;
	this->address = address;
}

ReaderSub::ReaderSub() {}

void ReaderSub::to_string() const {
	cout << "Номер читательского: " << num << "; \tФамилия: " << surname << "; Имя: " << name
		 << "; Отчество: " << second_name << "; Адрес: " << address << endl;
}